VSTestHost/TesteeTestAdapter.cs

```diff
@@ -129,19 +128,81 @@ class TesteeTestAdapter : MarshalByRefObject, ITestAdapter, IDisposable {
+                return value != null;
+            }
+            return runProps.TryGetValue(propertyName, out value);
+        }
+
         public void Run(ITestElement testElement, ITestContext testContext) {
             var testAdapter = GetAdapter(testElement.Adapter);
 
+            using (var screenRecorder = StartSceenRecorder(testElement, _runContext)) {
+                try {
+                    testAdapter.Run(testElement, testContext);
+                } catch (Exception ex) {
+                    var message = new TextTestResultMessage(
+                        _runContext.RunConfig.TestRun.Id,
+                        testElement,
+                        ex.ToString()
+                    );
+                    testContext.ResultSink.AddResult(message);
+                }
+
+                if (screenRecorder != null && !string.IsNullOrEmpty(screenRecorder.Failure)) {
+                    var message = new TextTestResultMessage(
+                        _runContext.RunConfig.TestRun.Id,
+                        testElement,
+                        screenRecorder.Failure
+                    );
+                    testContext.ResultSink.AddResult(message);
+                }
+            }
+        }
+
+        private ScreenRecorder StartSceenRecorder(ITestElement testElement, IRunContext runContext) {
+            string screenCapture;
+
+            if (!TryGetProperty(testElement, _runContext, VSTestProperties.ScreenCapture.Key, out screenCapture) ||
+                string.IsNullOrEmpty(screenCapture)) {
+                return null;
+            }
             try {
-                testAdapter.Run(testElement, testContext);
-            } catch (Exception ex) {
-                var message = new TextTestResultMessage(
-                    _runContext.RunConfig.TestRun.Id,
-                    testElement,
-                    ex.ToString()
-                );
-                testContext.ResultSink.AddResult(message);
+                if (!Path.IsPathRooted(screenCapture)) {
+                    screenCapture = Path.Combine(
+                        _runContext.RunConfig.TestRun.RunConfiguration.RunDeploymentOutDirectory,
+                        screenCapture
+                    );
+                }
+            } catch (ArgumentException) {
+                return null;
+            }
+
+            screenCapture = screenCapture.Replace("$id$", testElement.HumanReadableId);
+            screenCapture = screenCapture.Replace("$date$", DateTime.Today.ToShortDateString());
+            var screenRecorder = new ScreenRecorder(screenCapture);
+
+            string intervalString;
+            int interval;
+            if (TryGetProperty(testElement, runContext, VSTestProperties.ScreenCapture.IntervalKey, out intervalString) &&
+                int.TryParse(intervalString, out interval)) {
+                screenRecorder.Interval = TimeSpan.FromMilliseconds(interval);
```

Comment:
instructions in the readme say "The number of seconds between screenshots" but this is interpreting the value as milliseconds.